### ISSUE TITLE
h:inputFile + f:ajax  malformedXML: Top level node must be one of: changes, redirect, error, received: script instead.

### ISSUE THREAD

**Issue by pizzi80** [eclipse-ee4j/mojarra#4828]
Upload File with ajax support does not work as expected: 
it's not possible to "render" a portion of the page
I got a js alert with this message:

_"malformedXML: Top level node must be one of: changes, redirect, error, received: script instead."_

This happen because inside jsf.js @ line 2938 -> 2941:

`if (getParseErrorText(xml) !== PARSED_OK) {
    sendError(request, context, "malformedXML");
    return;
}`

in fact partialResponse.innerHTML: "<script/><changes><update id="preview"><![CDATA[<div id="preview" ....

innerHTML starts with _"<script/>"_

Additional BUG in JSF 3.0 does not fire the alert in Development mode
because it does not recognize the correct "mode" because inside jsf.js @ line 3063:

 if (typeof scripts[s].src === 'string' && scripts[s].src.match('\/javax\.faces\.resource\/jsf\.js\?.*ln=javax\.faces')) {

I think it's caused by "javax.faces" shoud be "jakarta.faces" ...

### TO REPRODUCE
**Java 11**:  Amazon Corretto 11.0.10 /  OpenJdk 11.0.9
**Tomcat**:   9.0.41   /  10.0
**JSF**:   2.3.14   /   3.0

**OS**: Linux Fedora 33 / Windows 10
**Browser**: Chrome 87 / 88 

**JSF Development Mode on web.xml**

```
@RequestScoped 
// or @ViewScoped
public class UploadImage {

	private Logger log = Logger.getLogger(UploadImage.class.getName());
	
	private Part file;									
	public Part getFile() { return file; }
	public void setFile(Part file) { this.file = file; }

	private String uploadedFile;
	public String getUploadedFile() { return uploadedFile; }

	public void upload() {
		uploadedFile = file.getSubmittedFileName();
		log.info("Uploaded File "+uploadedFile );
	}
}
```

```
<h:form id="upload" enctype="multipart/form-data">
  <h:inputFile id="image" value="#{uploadImage.file}">
    <f:ajax listener="#{uploadImage.upload}" render=":output" />
  </h:inputFile>
</h:form>

<h:panelGroup id="output" layout="block" style="border: 1px solid red; margin: 10px; padding:10px;">
  #{uploadImage.uploadedFile}
</h:panelGroup>
```

**Comment by manorrock** [comment]
Feel free to create a PR with a fix

**Comment by BalusC** [comment]
Curious. In order to exclude this to be caused by browser itself (or one of its extensions), can you please show the raw contents of the response?

E.g.

![image](https://user-images.githubusercontent.com/173372/106761489-4009f580-660b-11eb-93b3-6e742fdae536.png)

```
<?xml version='1.0' encoding='UTF-8'?>
<partial-response><changes><update id="preview"><![CDATA[<span id="preview">
			now: 1612362649546
			<br />
			part.submittedFileName: 89927e2f4bde24991649b353a37678b9.jpeg
		</span>]]></update><update id="j_id1:javax.faces.ViewState:0"><![CDATA[-1639785521564771570:236618271428335618]]></update></changes></partial-response>
```

Then we can advance based on whether the problematic `<script />` part is actually present in the response (thus added by JSF) or not (thus added by Chrome).

**Comment by pizzi80** [comment]
I've disable all my chrome extensions and the result is the same:

The raw response doesn't contain neither the `<script/>` tag
nor the modified part of html which should be the name of the uploaded file 

`<?xml version='1.0' encoding='UTF-8'?>
<partial-response>
  <changes>
    <update id="preview">
      <![CDATA[<div id="preview" style="border: 1px solid red; margin: 10px; padding: 10px;"></div>]]>
    </update>
  <update id="j_id1:jakarta.faces.ViewState:0">
    <![CDATA[7336769987308073180:-7703878573732739115]]>
  </update>
</changes>
</partial-response>`


In these 2 days I've inspected all the jsf.js code and made a naive patch to my jsf.js, 
but I'm not sure it's correct for a standard library like JSF, so I will not submit a PR.

@ line 2986 I've added this:

`// skip the misterious chrome script tag`
`if (responseType.nodeName === "script") {
  responseType = partialResponse.firstChild.nextSibling;
}`

and also made a replace to "javax" with "jakarta" to re-enable the
alert error like in JSF 2.3.14

After that, I've also made a lot of changes to JSF.js:

1. use strict
2. used const and let instead of generic var
3. removed IE detection and support
4. defined many global const 
5. rewritten many simple functions to a one liner const arrow function 

At the moment I'm working on replacing the FrameTransport with FormData 

If someone is interested I can donate it to mojarra, but I'm not sure that the community can accept 
all those changes ... let me know!
